Small improvements to docs. (#1500)
wasp-lang/wasp: web/docs/advanced/deployment/manually.md

There's a Dockerfile that defines an image for building the server in the `.wasp/build` directory.

> To run the server in production, deploy this Docker image to a hosting provider and ensure required environment variables on the provider are correctly set up (mechanism of setting these up is specific per provider).

Reviewer: ```suggestion
To run the server in production, deploy this Docker image to a hosting provider and ensure the required environment variables on the provider are correctly set up (the mechanism of setting these up is specific per provider).
```

All necessary environment variables are listed in the next section.

#### Environment Variables

Here are the environment variables your server will be looking for:

- `DATABASE_URL` (required)

  The URL of the Postgres database you want your app to use (e.g., `postgresql://mydbuser:mypass@localhost:5432/nameofmydb`).

> - `WASP_WEB_CLIENT_URL` (required)

Reviewer: We agreed on using the `<Required />` component

`import { Required } from '@site/src/components/Required'`

Author: Ok, I did the change, but just a note: this is not an API reference, instead this is listing env variables that server can expect.

  The URL where you plan to deploy your frontend app is running (e.g., `https://<app-name>.netlify.app`).
  The server needs to know about it to properly configure Same-Origin Policy (CORS) headers.

- `JWT_SECRET` (required if using Wasp Auth)

  You only need this environment variable if you're using Wasp's `auth` features.
  Set it to a random string at least 32 characters long (you can use an [online generator](https://djecrety.ir/)).

> - `PORT` (optional)

Reviewer: We agreed on not writing `(optional)` if it's optional 🤷‍♂️ we didn't do it in other parts of the docs

There's a Dockerfile that defines an image for building the server in the `.wasp/build` directory.

To run the server in production, deploy this Docker image to a hosting provider and ensure the required environment variables on the provider are correctly set up (the mechanism of setting these up is specific per provider).
All necessary environment variables are listed in the next section.

#### Environment Variables

Here are the environment variables your server will be looking for:

- `DATABASE_URL` (<Required />)

  The URL of the Postgres database you want your app to use (e.g., `postgresql://mydbuser:mypass@localhost:5432/nameofmydb`).

> - `WASP_WEB_CLIENT_URL` (<Required />)

Author: Makes sense but what about one that has "if using Wasp Auth"?

> - `WASP_WEB_CLIENT_URL` (<Required />)

Author: Ok what I did is remove `()` from all of them except for the last one, since I need some way to group that ` if using Wasp auth` with it.

- Railway (server, client and database)
- Heroku (server and database)

## Fly.io (server and database)

> We will show how to deploy Wasp server and provision a database for it on Fly.io.

Reviewer: ```suggestion
We will show how to deploy the server and provision a database for it on Fly.io.
```

1. Run `flyctl config save -a <app-name>` to regenerate the `fly.toml` file from the remote state stored in Fly.io.

## Netlify (client)

> We will show how to deploy Wasp client on Netlify.

Reviewer: ```suggestion
We'll show how to deploy the Wasp client on Netlify.
```

Reviewer: There isn't a thing as "Wasp client" IMHO

Make sure you set this URL as the `WASP_WEB_CLIENT_URL` environment variable in your server hosting environment (e.g., Fly.io or Heroku).
:::

## Railway (server, client and database)

> We will show how to deploy Wasp client, Wasp server, and provision a database on Railway.

Reviewer: ```suggestion
We will show how to deploy the client, the server, and provision a database on Railway.
```

- run `railway up` in the `.wasp/build` directory (server)
- repeat all the steps in the `.wasp/build/web-app` directory (client)

## Heroku (server and database)

> We will show how to deploy Wasp server and provision a database for it on Heroku.

Reviewer: ```suggestion
We will show how to deploy the server and provision a database for it on Heroku.
```
